tests/std/tests/P0627R6_unreachable/test.cpp

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <cassert>
+#include <utility>
+
+constexpr int test_impl(int arg) {
+    switch (arg) {
+    case 1:
+        return 'x';
+
+    case 2:
+        return 'y';
+
+    default:
+        std::unreachable();
+    }
+}
+
+constexpr bool test() {
+    assert(test_impl(1) == 'x');
+    assert(test_impl(2) == 'y');
+    return true;

```

Comment:
Should this not test the noexceptness?
```suggestion
assert(noexcept(std::unreachable())); // strengthened
return true;
```

Reply:
I don't know. 
I think we avoid testing implementation details, unless necessary.
I'm not sure if this case is "necessary".

Reply:
I suppose it wouldn't hurt - although it should be a `static_assert`.